### ISSUE TITLE
[Linux]Launcher will error out at launch after the first run

### ISSUE THREAD

**Issue by Unaccounted4** [SoapboxRaceWorld/GameLauncher_NFSW#26]
- **Launcher Version:** 2.0.11.0 (build D0246F0)
- **Operating system:** Ubuntu 18.04.4
- **Antivirus/Firewall (if any):** firewalld
- **Wine/Mono Version (if any):** Wine 5.2 Staging (32bit Prefix) / No Mono, winetricks dotnet46

After I've run it the first time and quit, attempting to reopen the launcher causes it to throw up an error about "object reference not set to an instance of an object". Here's the full log:
```
Application does not support just-in-time (JIT)
debugging. See the end of this message for details.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at GameLauncher.MainScreen..ctor()
   at GameLauncher.Program.Main()


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
GameLauncher
    Assembly Version: 2.0.11.0
    Win32 Version: 2.0.11.0
    CodeBase: file:///C:/NFSW/GameLauncher.exe
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
INIFileParser
    Assembly Version: 2.5.2.0
    Win32 Version: 2.5.2
    CodeBase: file:///C:/NFSW/INIFileParser.DLL
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Nancy.Hosting.Self
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.0.0
    CodeBase: file:///C:/NFSW/Nancy.Hosting.Self.DLL
----------------------------------------
Nancy
    Assembly Version: 2.0.0.0
    Win32 Version: 2.0.0.0
    CodeBase: file:///C:/NFSW/Nancy.DLL
----------------------------------------
System.Net.Http
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.6.81.0 built by: NETFXREL2
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_32/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
Microsoft.CSharp
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll
----------------------------------------
System.Dynamic
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.81.0
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------
DiscordRPC
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/NFSW/DiscordRPC.DLL
----------------------------------------
Microsoft.WindowsAPICodePack.Shell
    Assembly Version: 1.1.0.0
    Win32 Version: 1.1.0.0
    CodeBase: file:///C:/NFSW/Microsoft.WindowsAPICodePack.Shell.DLL
----------------------------------------
Microsoft.WindowsAPICodePack
    Assembly Version: 1.1.0.0
    Win32 Version: 1.1.0.0
    CodeBase: file:///C:/NFSW/Microsoft.WindowsAPICodePack.DLL
----------------------------------------
Newtonsoft.Json
    Assembly Version: 12.0.0.0
    Win32 Version: 12.0.3.23909
    CodeBase: file:///C:/NFSW/Newtonsoft.Json.DLL
----------------------------------------

************** JIT Debugging **************
Application does not support Windows Forms just-in-time (JIT)
debugging. Contact the application author for more
information.
```

There's also an issue with the game freezing after a minute or so of being in game proper, but I'm not sure if this issue tracker is for that.

**Comment by ghost** [comment]
Yup, i were able to reproduce that, and i am working on a fix. however, linux was never supposed to be supported, but... i will make it work.
Testing using wine 5.0 and wine-mono.

**Comment by ghost** [comment]
Possible cause: Mouse cursor is being loaded, which is not really supported on linux

**Comment by Unaccounted4** [comment]
To be honest, the only reason I submitted it was because I saw the mentions of WINE in the template.

Thank you!

**Comment by ghost** [comment]
Yup, it was supposed to get an update that would support linux, but we always had one issue making Downloader working due to EA code issues.

**Comment by ghost** [comment]
![Zrzut ekranu Deepin_20200222230646](https://user-images.githubusercontent.com/6225233/75099996-6d12f300-55c8-11ea-9f5c-eb3addf1dfb1.png)
okay managed to make launcher work under Deepin distro using wine 5.0 and wine-mono

**Comment by Unaccounted4** [comment]
Oh, nice! Thank you!

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/6225233/75298138-f6792e00-5831-11ea-87ee-ba4c9479d60c.png)
forgot to add that screen few hours later, already ingame. TODO: fix chat that freezes whole game.

**Comment by Unaccounted4** [comment]
Oh, it's the chat that's doing it! I was wondering...

Let me know if you figure it out. I already tried installing corefonts in case it was a missing font but it didn't make a difference.

**Comment by ghost** [comment]
Yea, the chat can be open for approx. 10 seconds, then something freezes whole game except audio.

**Comment by Unaccounted4** [comment]
I see you have a Paypal pool going. I'll kick you a few ££ if you can figure it out and fix it.

**Comment by ghost** [comment]
[LinuxLauncher.zip](https://github.com/SoapboxRaceWorld/GameLauncher_NFSW/files/4271196/LinuxLauncher.zip)
Alright, this is latest built launcher, that "seems" to launch without issues, game wasnt yet tested.

**Comment by ghost** [comment]
Latest update has game working now too, except first error that can be ignored.

**Comment by Unaccounted4** [comment]
Alright, I've downloaded the new 2.0.11.1 and on the first run it stays stuck at checking the list of files to download. Closing and reopening allows the Launcher to download the game files. 

Once download finishes and I hit Play I get this error:
```
See the end of this message for details on invoking \njust-in-time (JIT) debugging instead of this dialog box.\n\n************** Exception Text **************\nSystem.ComponentModel.Win32Exception (0x80004005): ApplicationName='/bin/wine', CommandLine='Z:\home\unaccounted4\.wine\drive_c\Release\GameFiles/nfsw.exe WUGG http://127.0.0.1:7945/nfsw/Engine.svc 03715b3e-0d0e-4f54-98ce-453031502f77 23134', CurrentDirectory='Z:\home\unaccounted4\.wine\drive_c\Release\GameFiles', Native error= File not found.

  at System.Diagnostics.Process.StartWithCreateProcess (System.Diagnostics.ProcessStartInfo startInfo) [0x0029f] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at System.Diagnostics.Process.Start () [0x0003a] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.Start()
  at System.Diagnostics.Process.Start (System.Diagnostics.ProcessStartInfo startInfo) [0x0001b] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at GameLauncher.MainScreen.LaunchGame (System.String userId, System.String loginToken, System.String serverIp, System.Windows.Forms.Form x) [0x000ff] in <499e26b1fc41485d8829430319cfe25f>:0 
  at GameLauncher.MainScreen+<>c__DisplayClass111_0.<StartGame>b__0 () [0x0002c] in <499e26b1fc41485d8829430319cfe25f>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <a1336bab325642fd867e87ca7626bada>:0 

\n************** Loaded Assemblies **************\nmscorlib\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/4.5/mscorlib.dll\n----------------------------------------\nGameLauncher\n    Assembly Version: 2.0.11.1\n    Win32 Version: 2.0.11.1\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/GameLauncher.exe\n----------------------------------------\nSystem\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System/4.0.0.0__b77a5c561934e089/System.dll\n----------------------------------------\nSystem.Windows.Forms\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Windows.Forms/4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll\n----------------------------------------\nINIFileParser\n    Assembly Version: 2.5.2.0\n    Win32 Version: 2.5.2\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/INIFileParser.dll\n----------------------------------------\nNancy.Hosting.Self\n    Assembly Version: 2.0.0.0\n    Win32 Version: 2.0.0.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Nancy.Hosting.Self.dll\n----------------------------------------\nSystem.Drawing\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Drawing/4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll\n----------------------------------------\nAccessibility\n    Assembly Version: 4.0.0.0\n    Win32 Version: \n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Accessibility/4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll\n----------------------------------------\nDiscordRPC\n    Assembly Version: 1.0.150.0\n    Win32 Version: 1.0.150.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/DiscordRPC.dll\n----------------------------------------\nSystem.Configuration\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Configuration/4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll\n----------------------------------------\nSystem.Xml\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Xml/4.0.0.0__b77a5c561934e089/System.Xml.dll\n----------------------------------------\nMono.Security\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Mono.Security/4.0.0.0__0738eb9f132ed756/Mono.Security.dll\n----------------------------------------\nI18N\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/I18N/4.0.0.0__0738eb9f132ed756/I18N.dll\n----------------------------------------\nI18N.West\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/I18N.West/4.0.0.0__0738eb9f132ed756/I18N.West.dll\n----------------------------------------\nNancy\n    Assembly Version: 2.0.0.0\n    Win32 Version: 2.0.0.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Nancy.dll\n----------------------------------------\nSystem.Core\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Core/4.0.0.0__b77a5c561934e089/System.Core.dll\n----------------------------------------\nSystem.Management\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Management/4.0.0.0__b03f5f7f11d50a3a/System.Management.dll\n----------------------------------------\nSystem.Net.Http\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Net.Http/4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll\n----------------------------------------\nFlurl\n    Assembly Version: 2.8.2.0\n    Win32 Version: 2.8.2.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Flurl.dll\n----------------------------------------\nFlurl.Http\n    Assembly Version: 2.4.2.0\n    Win32 Version: 2.4.2.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Flurl.Http.dll\n----------------------------------------\nMicrosoft.WindowsAPICodePack\n    Assembly Version: 1.1.0.0\n    Win32 Version: 1.1.0.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Microsoft.WindowsAPICodePack.dll\n----------------------------------------\nAnonymously Hosted DynamicMethods Assembly\n    Assembly Version: 0.0.0.0\n    Win32 Version: n/a\n    CodeBase: \n----------------------------------------\nMicrosoft.CSharp\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Microsoft.CSharp/4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll\n----------------------------------------\nMicrosoft.WindowsAPICodePack.Shell\n    Assembly Version: 1.1.0.0\n    Win32 Version: 1.1.0.0\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Microsoft.WindowsAPICodePack.Shell.dll\n----------------------------------------\nPresentationFramework\n    Assembly Version: 3.0.0.0\n    Win32 Version: 3.0.6920.5738 built by: QFE\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/PresentationFramework.dll\n----------------------------------------\nPresentationCore\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.8.4121.0 built by: NET48REL1LAST_C\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/PresentationCore.dll\n----------------------------------------\nWindowsBase\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/WindowsBase/4.0.0.0__31bf3856ad364e35/WindowsBase.dll\n----------------------------------------\nNewtonsoft.Json\n    Assembly Version: 12.0.0.0\n    Win32 Version: 12.0.3.23909\n    CodeBase: file:///Z:/home/unaccounted4/.wine/drive_c/Release/Newtonsoft.Json.dll\n----------------------------------------\nSystem.Numerics\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Numerics/4.0.0.0__b77a5c561934e089/System.Numerics.dll\n----------------------------------------\nSystem.Runtime.Serialization\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Runtime.Serialization/4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll\n----------------------------------------\nSystem.Data\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Data/4.0.0.0__b77a5c561934e089/System.Data.dll\n----------------------------------------\n\n************** JIT Debugging **************\n


```

**Comment by ghost** [comment]
Yea, thats the error i am talking about, this has not yet been fixed, but the game launches. The fix is in progress

**Comment by ghost** [comment]
Oh wait, that's "wine" error, can you check if u have `/bin/wine` file there?

**Comment by Unaccounted4** [comment]
Oh! I'm using Wine staging, it's located in `/opt/wine-staging/bin`

**Comment by ghost** [comment]
Alright, i will create some sort of detecting wine location, seems to have like... a lot of them, since `wine` itself fails

**Comment by ghost** [comment]
Alright, latest executable seems to load launcher and game without issues (except chat, which i have also patched, but need downloader)

**Comment by Unaccounted4** [comment]
With the latest version, I still have to quit out the first time for it to download but afterwards I can launch the game and drive around without opening the chat.

I also collected all the daily gems and it worked as expected. Super looking forward to the chat fix!

![soapbox](https://user-images.githubusercontent.com/45717994/75716935-185a3100-5cc8-11ea-97ac-6c9acfdbe4d7.png)

**Comment by ghost** [comment]
Nice to see that, i am already working on `wine-nfsw.deb` that will work for both launcher and game. This is already tested with older wine (1.7), but would be nicer to use 5.0 instead :D 

I sould test it on proton, maybe has fixes patched.

**Comment by Unaccounted4** [comment]
Proton does have a bunch of patches applied. Did you have to patch WINE to stop the chat from freezing the game?

Also, it may be related to me keeping chat disabled or something to do with races having their own chat channel, but I couldn't join multiplayer races. The game locked up in the loading screen.

**Comment by ghost** [comment]
That's also an issue, since modloader doesnt properly works with game. separate fix is also comming for that

**Comment by ghost** [comment]
Well. an update. Everything works as expected:
https://imgur.com/a/qVpetcU

New wine files: https://cdn.discordapp.com/attachments/684929550366670853/685595356028731416/compiled.7z (wine-nfsw.deb is comming with that)
Lutris users can extract it under `~/.local/share/lutris/runners/wine` to select new wine directly from there

For multiplayer fixes and its modloader, we need to set up new overrides, by using regedit (launcher will do that in next update):
```
[HKEY_CURRENT_USER\Software\Wine\DllOverrides]
"dinput8"="native,builtin"
"lightfx"="native,builtin"
```

First one is for new system (ModLoader), second one is for crc bypass used in ModManager instead.

More info will come as soon i will make new packages for that.

EDIT: An info from one of the testers:
`dll overrides can be set inside lutris game config->Runner options->DLL overrides`

**Comment by Unaccounted4** [comment]
Nice!

So, what patches did you do to WINE?

EDIT: Looks like you hard-linked some libraries, can't redistribute it like that.
```
winecfg
wine: failed to initialize: /home/metonator/compiled/lib/wine/ntdll.dll.so: cannot open shared object file: No such file or directory
```

**Comment by ghost** [comment]
Oh, will fix it
EDIT:
Thinking to use `/opt/wine-nfsw` instead.

**Comment by metonator** [comment]
QUICK REPLY:
Thinking to use `/opt/wine-nfsw` instead.

**Comment by wagim47274** [comment]
Hey, I'm using a tempmail, so you will probably not get a response,,to any questions (sorry), I am using linux also, I've got the patch and when i start the launcher it says "Launcher status - Linux fix" at the bottom left corner, everything is fine up to the point of pressing "Play now", all of the files are downloading, but when the message "Loading game, minimizing in..." appears an error appears. If i don't install the linux patch the game just closes when I press "Play now", with the patch the message appears:

```
See the end of this message for details on invoking \njust-in-time (JIT) debugging instead of this dialog box.\n\n************** Exception Text **************\nSystem.ComponentModel.Win32Exception (0x80004005): ApplicationName='wine', CommandLine='Z:\Main\Games\NFSW\GameFiles/nfsw.exe NIGHTRIDERZ http://127.0.0.1:8233/nfsw/Engine.svc c3e19bbc-938e-4cba-90f8-3f3d8a4135d0 165487', CurrentDirectory='', Native error= File not found.

  at System.Diagnostics.Process.StartWithCreateProcess (System.Diagnostics.ProcessStartInfo startInfo) [0x0029f] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at System.Diagnostics.Process.Start () [0x0003a] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.Start()
  at System.Diagnostics.Process.Start (System.Diagnostics.ProcessStartInfo startInfo) [0x0001b] in <f508ff7dc2d3475abfc25b6b60600edf>:0 
  at GameLauncher.MainScreen.LaunchGame (System.String userId, System.String loginToken, System.String serverIp, System.Windows.Forms.Form x) [0x000e9] in <f0f9a75d8acf436ba3726aefd858bded>:0 
  at GameLauncher.MainScreen+<>c__DisplayClass111_0.<StartGame>b__0 () [0x0002c] in <f0f9a75d8acf436ba3726aefd858bded>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <a1336bab325642fd867e87ca7626bada>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <a1336bab325642fd867e87ca7626bada>:0 

\n************** Loaded Assemblies **************\nmscorlib\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/4.5/mscorlib.dll\n----------------------------------------\nGameLauncher\n    Assembly Version: 2.0.11.0\n    Win32 Version: 2.0.11.0\n    CodeBase: file:///Z:/Main/Games/NFSW/GameLauncher.exe\n----------------------------------------\nSystem\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System/4.0.0.0__b77a5c561934e089/System.dll\n----------------------------------------\nSystem.Windows.Forms\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Windows.Forms/4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll\n----------------------------------------\nINIFileParser\n    Assembly Version: 2.5.2.0\n    Win32 Version: 2.5.2\n    CodeBase: file:///Z:/Main/Games/NFSW/INIFileParser.dll\n----------------------------------------\nNancy.Hosting.Self\n    Assembly Version: 2.0.0.0\n    Win32 Version: 2.0.0.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Nancy.Hosting.Self.dll\n----------------------------------------\nSystem.Drawing\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Drawing/4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll\n----------------------------------------\nAccessibility\n    Assembly Version: 4.0.0.0\n    Win32 Version: \n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Accessibility/4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll\n----------------------------------------\nDiscordRPC\n    Assembly Version: 1.0.150.0\n    Win32 Version: 1.0.150.0\n    CodeBase: file:///Z:/Main/Games/NFSW/DiscordRPC.dll\n----------------------------------------\nSystem.Configuration\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Configuration/4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll\n----------------------------------------\nSystem.Xml\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Xml/4.0.0.0__b77a5c561934e089/System.Xml.dll\n----------------------------------------\nMono.Security\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Mono.Security/4.0.0.0__0738eb9f132ed756/Mono.Security.dll\n----------------------------------------\nI18N\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/I18N/4.0.0.0__0738eb9f132ed756/I18N.dll\n----------------------------------------\nI18N.West\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.0.0.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/I18N.West/4.0.0.0__0738eb9f132ed756/I18N.West.dll\n----------------------------------------\nNancy\n    Assembly Version: 2.0.0.0\n    Win32 Version: 2.0.0.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Nancy.dll\n----------------------------------------\nSystem.Core\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Core/4.0.0.0__b77a5c561934e089/System.Core.dll\n----------------------------------------\nSystem.Management\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Management/4.0.0.0__b03f5f7f11d50a3a/System.Management.dll\n----------------------------------------\nSystem.Net.Http\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Net.Http/4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll\n----------------------------------------\nFlurl\n    Assembly Version: 2.8.2.0\n    Win32 Version: 2.8.2.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Flurl.dll\n----------------------------------------\nFlurl.Http\n    Assembly Version: 2.4.2.0\n    Win32 Version: 2.4.2.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Flurl.Http.dll\n----------------------------------------\nMicrosoft.WindowsAPICodePack\n    Assembly Version: 1.1.0.0\n    Win32 Version: 1.1.0.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Microsoft.WindowsAPICodePack.dll\n----------------------------------------\nAnonymously Hosted DynamicMethods Assembly\n    Assembly Version: 0.0.0.0\n    Win32 Version: n/a\n    CodeBase: \n----------------------------------------\nMicrosoft.CSharp\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/Microsoft.CSharp/4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll\n----------------------------------------\nMicrosoft.WindowsAPICodePack.Shell\n    Assembly Version: 1.1.0.0\n    Win32 Version: 1.1.0.0\n    CodeBase: file:///Z:/Main/Games/NFSW/Microsoft.WindowsAPICodePack.Shell.dll\n----------------------------------------\nPresentationFramework\n    Assembly Version: 3.0.0.0\n    Win32 Version: 3.0.6920.5738 built by: QFE\n    CodeBase: file:///Z:/Main/Games/NFSW/PresentationFramework.dll\n----------------------------------------\nPresentationCore\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.8.4121.0 built by: NET48REL1LAST_C\n    CodeBase: file:///Z:/Main/Games/NFSW/PresentationCore.dll\n----------------------------------------\nWindowsBase\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/WindowsBase/4.0.0.0__31bf3856ad364e35/WindowsBase.dll\n----------------------------------------\nNewtonsoft.Json\n    Assembly Version: 12.0.0.0\n    Win32 Version: 12.0.3.23909\n    CodeBase: file:///Z:/Main/Games/NFSW/Newtonsoft.Json.dll\n----------------------------------------\nSystem.Numerics\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Numerics/4.0.0.0__b77a5c561934e089/System.Numerics.dll\n----------------------------------------\nSystem.Runtime.Serialization\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Runtime.Serialization/4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll\n----------------------------------------\nSystem.Data\n    Assembly Version: 4.0.0.0\n    Win32 Version: 4.6.57.0\n    CodeBase: file:///C:/windows/mono/mono-2.0/lib/mono/gac/System.Data/4.0.0.0__b77a5c561934e089/System.Data.dll\n----------------------------------------\n\n************** JIT Debugging **************\n


```

**Comment by ghost** [comment]
Try updating launcher. i think i have fixed that issue lastly
However: there's new wine http://cdn.mtntr.pl/wine/wine-nfsw.zip
Extract to `/opt` folder (`sudo mv ~/Downloads/wine-nfsw.zip /opt && cd /opt && sudo unzip wine-nfsw.zip` or smth like that)

**Comment by Unaccounted4** [comment]
Forcing people to extract an application to a system folder is not good practice.

**Comment by wagim47274** [comment]
Not a good idea but hey, atlest there's somebody thats caring. I've reinstalled the game, and now I don't get an exception, but when it says "minimizing after..." the game closes itself, I see it's starting another process with the server name (tried nightriders and zesty), but after that all of them terminate (the launcher and nfsw.exe). Moved the files that you've said to the folder /opt. Anything else I can do to not make the applications shut down ?

**Comment by mirkobrombin** [comment]
> Forcing people to extract an application to a system folder is not good practice.

The `/opt` directory can be used for this context.
Obviously the user must take into consideration the reliability of the developer. However, extracting it to `/home` can also be dangerous.

**Comment by mirkobrombin** [comment]
> Try updating launcher. i think i have fixed that issue lastly
> However: there's new wine http://cdn.mtntr.pl/wine/wine-nfsw.zip
> Extract to `/opt` folder (`sudo mv ~/Downloads/wine-nfsw.zip /opt && sudo unzip wine-nfsw.zip` or smth like that)

With this version I can see the other players and use the chat. I will create an installer for lutris to automate the process.

**Comment by ghost** [comment]
> > Try updating launcher. i think i have fixed that issue lastly
> > However: there's new wine http://cdn.mtntr.pl/wine/wine-nfsw.zip
> > Extract to `/opt` folder (`sudo mv ~/Downloads/wine-nfsw.zip /opt && sudo unzip wine-nfsw.zip` or smth like that)
> 
> With this version I can see the other players and use the chat. I will create an installer for lutris to automate the process.

There's already one in creation by discord user named Guldoman#1845, can be found on both SBRW and WUGG discord

Best option now would be to create `.deb` (Debian, Ubuntu, Deepin, ...), `.rpm` (Fedora, Mandriva, SUSE, ...), and `.pkg.tar.gz` (Arch Linux) files for non-Lutris packages. 
Currenly working on `.deb` one.

**Comment by mirkobrombin** [comment]
Oh good

**Comment by Guldoman** [comment]
Lutris users can use this installer: http://cdn.mtntr.pl/nfsw_installer.yml
Download it, open a terminal in the same folder and run `lutris -i $PWD/nfsw_installer.yml`.
Because of a lutris bug (lutris/lutris#2644), automatically setting the wine version to `custom` doesn't work, so when the installer finishes right click on the game->Configure->Runner options->Wine version->Custom; then click Save and run the game.

Should I publish the installer on the Lutris website, even if this bug persists?

**Comment by ghost** [comment]
I would say is ok, the bug should be fixed on next release (i think)

**Comment by Unaccounted4** [comment]
That Lutris installer has the same issue with it looking for the libraries in /home/metonator. The wine-nsfw build is still hardlinked.

```
Running ~/Jogos/nfsw-sbrw/wine-nfsw/bin/wine ~/Jogos/nfsw-sbrw/GameLauncher.exe
wine: failed to initialize: /home/metonator/compiled/lib/wine/ntdll.dll.so: cannot open shared object file: No such file or directory
Waiting on children
Exit with returncode 1
2020-03-08 20:39:08,550: Game still running (state: running)
2020-03-08 20:39:08,551: Stopping Soapbox Race World (wine)
```

**Comment by wagim47274** [comment]
Hello again everybody, I found a fix that got the game working, when I get in the game it wants to set the chat language or channel, when I set it it works for 5 seconds and crashes, the fix ? The fix was simple, press the button that enables the chat and go to the safehouse (upper left button), wait for a little bit to load and get back in freeroam, it should not crash now. Hope I helped somebody that is/was experiencing the same issue.

**Comment by ghost** [comment]
> That Lutris installer has the same issue with it looking for the libraries in /home/metonator. The wine-nsfw build is still hardlinked.
> 
> ```
> Running ~/Jogos/nfsw-sbrw/wine-nfsw/bin/wine ~/Jogos/nfsw-sbrw/GameLauncher.exe
> wine: failed to initialize: /home/metonator/compiled/lib/wine/ntdll.dll.so: cannot open shared object file: No such file or directory
> Waiting on children
> Exit with returncode 1
> 2020-03-08 20:39:08,550: Game still running (state: running)
> 2020-03-08 20:39:08,551: Stopping Soapbox Race World (wine)
> ```

Trying to figure out this. I am installing 3 extra distros to verify that (Arch, Ubuntu, Fedora).

**Comment by ghost** [comment]
Ok, new wine is here: https://cdn.discordapp.com/attachments/684929550366670853/686338022932873244/wine-nfsw.tar.xz
hardlinked libs should be gone (as always, `/opt` directory)

**Comment by Unaccounted4** [comment]
Yup, new build is checking for /opt

```
Running ~/Jogos/nfsw-sbrw/wine-nfsw/bin/wine ~/Jogos/nfsw-sbrw/GameLauncher.exe
wine: failed to initialize: /opt/wine-nfsw/lib/wine/ntdll.dll.so: cannot open shared object file: No such file or directory
Waiting on children
Exit with returncode 1
2020-03-08 22:31:08,898: Game still running (state: running)
2020-03-08 22:31:08,898: Stopping Soapbox Race World (wine)
```

**Comment by ghost** [comment]
Still hardlinking... lemme read carefully now wine building to avoid that.

**Comment by Unaccounted4** [comment]
That would be perfect! And @Guldoman, you absolutely should submit that script to Lutris, once the issue with this custom wine build is sorted.

**Comment by Guldoman** [comment]
@Unaccounted4, do you have standard wine installed? Could you try running the game via `lutris --debug` and attaching the log here?

**Comment by Unaccounted4** [comment]
I do have standard WINE Staging 5.3 installed. Changing Lutris to use system WINE allows the launcher to start.

Changing to the custom WINE here's the log: https://gist.github.com/Unaccounted4/c98290a7e0f030b579792e7534872a59

**Comment by ghost** [comment]
Ok, i've contacted moderation team for custom wine patch inside lutris runtime api, hoping they will reply asap

**Comment by Unaccounted4** [comment]
I found another issue 

Since you built WINE with GLibC 2.28 it doesn't work on 18.04, as it still only has 2.27. I set a chroot so the launcher would see the wine-nsfw folder as if it were in /opt

```
Running /opt/wine-nfsw/bin/wine /home/unaccounted4/Jogos/nfsw-sbrw/GameLauncher.exe
wine: failed to initialize: /lib/i386-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/wine-nfsw/lib/wine/ntdll.dll.so)
Waiting on children
Exit with returncode 1
```

**Comment by ghost** [comment]
```
echo "deb http://ftp.de.debian.org/debian sid main" >> sudo tee /etc/apt/sources.list.d/debian.list
sudo apt update
sudo apt install glibc
```

This should fix the problem, since i used debian 10 to compile all the stuff.

**Comment by ghost** [comment]
https://lutris.net/games/soapbox-race-world-need-for-speed-world/ now on lutris. with all fixes.

**Comment by melroy89** [comment]
@Unaccounted4 Thank you so much for the Unix version. You don't get enough appreciation for this :)

**Comment by Unaccounted4** [comment]
I'm just the one who figured he might as well raise the issue and test whenever there's a need for it. The big kudos are for @Zacam, @DavidCarbon, and Metonator. They did all the heavy lifting.

**Comment by melroy89** [comment]
I might need some help from them in that case, see: https://youtu.be/gpjJ36Lj4BY

I'm using Wine v8.0.2 and installed dinput8, which seem actually load the mod but crashes the game (and installed DXVK, with or without doesn't matter in this case). I tried Windows XP 32-bit and Windows 7 32-bit bottles.

[SBRCrashDump_CL0_2023.10.08_17.33.12.txt](https://github.com/SoapboxRaceWorld/GameLauncher_NFSW/files/12841306/SBRCrashDump_CL0_2023.10.08_17.33.12.txt)

Launcher itself seems to work fine, I can download the game and login.

**Comment by Unaccounted4** [comment]
You can use this installer with Lutris: https://lutris.net/games/need-for-speed-world/

It should set everything up correctly and the game should work just fine.
